src/Adapter/Product/Image/ProductImagePathFactory.php

```diff
@@ -90,4 +155,20 @@ public function getHelperThumbnail(int $productId, int $shopId): string
     {
         return sprintf('%sproduct_mini_%d_%d.jpg', $this->temporaryImgDir, $productId, $shopId);
     }
+
+    /**
+     * @param Image $image
+     *
+     * @return string
+     */
+    private function getBaseImagePathWithoutExtension(Image $image): string
+    {
+        if ($this->isLegacyImageMode) {
+            $path = $image->id_product . '-' . $image->id;
+        } else {
+            $path = ltrim($image->getImgPath(), '/');
```

Comment:
This part could be handled internally, no need to use lagcy code to get image folder
It can be built based on its ID alone

Reply:
if you check the method has more code, and uses couple more methods inside, I really doubt I need to reinvent the bike here  :sweat: (unless I didn't understand your request)